lib/macos.h

```diff
@@ -25,7 +25,8 @@
  ***************************************************************************/
 #include "curl_setup.h"
 
-#if defined(__APPLE__) && (!defined(TARGET_OS_OSX) || TARGET_OS_OSX)
+#if defined(__APPLE__) && TARGET_OS_MAC && \
```

Comment:
The c-ares build is failing with:

```
In file included from easy.c:66:
../lib/macos.h:28:27: error: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Werror,-Wundef]
#if defined(__APPLE__) && TARGET_OS_MAC && \
                          ^
```

I was expecting <TargetConditionals.h> to have been included by curl_setup.h but it only does that when not building for c-ares.

Maybe the nicest thing to do is to have curl_setup.h always include it (on Darwin). lib/vtls/sectransp.c also uses `TARGET_OS_` macros without first including <TargetConditionals.h>, probably expecting it to be included by the other Darwin headers it's already including.

Alternately, I can include <TargetConditionals.h> everywhere a `TARGET_OS_` macro is used.

Reply:
> I was expecting <TargetConditionals.h> to have been included by curl_setup.h but it only does that when not building for c-ares.

I suppose because the `SCDynamicStoreCopyProxies` thingie is not needed for builds that use c-ares since then it does not use the regular macOS resolver functions.